sdk/eventhubs/azure-messaging-eventhubs/src/producer_client.cpp

```diff
@@ -89,6 +89,16 @@ namespace Azure { namespace Messaging { namespace EventHubs {
       connection.second.Close(context);
     }
 #endif
+    // Remove all the sessions and connections after they've been closed.
+    while (!m_sessions.empty())
+    {
+      m_sessions.erase(m_sessions.begin());
+    }
+
+    while (!m_connections.empty())
+    {
+      m_connections.erase(m_connections.begin());
+    };
```

Comment:
If you agree with suggestions, I think this pattern is used in a few other places in the SDK, we can update them too.
If you dont agree with suggestions, and there's something I missed - there is an unnecessary `;` after the second `while`'s `}` (line 101)
```suggestion
    // Remove all the sessions and connections after they've been closed.
    m_sessions.clear();
    m_connections.clear();
```